testsuites/CBLTester/system_tests/test_system_concurrent_testing.py

```diff
@@ -0,0 +1,384 @@
+            # Query results do not store in memory for dot net, so no need to release memory for dotnet
+            if platform.lower() not in ("net-msft", "uwp", "xamarin-ios", "xamarin-android"):
+                _releaseQueryResults(base_url, results)
+
+            _check_doc_count(db_obj, cbl_db)
+            # removing ids of deleted doc from the list
+            doc_ids = doc_ids - docs_to_delete
+
+            #############################
+            # Creating docs on CBL side #
+            #############################
+            name = db_obj.getName(cbl_db)
+            docs_to_create = ["{}_doc_{}".format(name, doc_id) for doc_id in range(doc_id_for_new_docs, doc_id_for_new_docs + num_of_docs_to_add)]
+            added_docs = {}
+            new_doc_ids = []
+            for doc_id in docs_to_create:
+                if generator == "complex_doc":
+                    data = complex_doc()
+                elif generator == "four_k":
+                    data = four_k()
+                elif generator == "simple_user":
+                    data = simple_user()
+                else:
+                    data = simple()
+                data["channels"] = sg_channels
+                data["id"] = doc_id
+                added_docs[doc_id] = data
+                new_doc_ids.append(doc_id)
+            doc_ids.update(new_doc_ids)
+            _log_system_test("Thread {}".format(db_name),
+                             "docs creating on CBL",
+                             "creating {} docs on {} - {}".format(len(docs_to_create), db_obj.getName(cbl_db), new_doc_ids))
+            db_obj.saveDocuments(cbl_db, added_docs)
+            time.sleep(5)
+
+            _replicaton_status_check(thread_name, repl_obj, repl, repl_status_check_sleep_time)
+
+            results = query.query_get_docs_limit_offset(cbl_db, limit=query_limit,
+                                                        offset=query_offset)
+            # Query results do not store in memory for dot net, so no need to release memory for dotnet
+            if platform.lower() not in ("net-msft", "uwp", "xamarin-ios", "xamarin-android"):
+                _releaseQueryResults(base_url, results)
+
+            time.sleep(5)
+            doc_id_for_new_docs += num_of_docs_to_add
+            query_offset += query_limit
+            _check_doc_count(db_obj, cbl_db)
+
+            current_time = datetime.now()
+
+        # stopping replication
+        _log_system_test("Thread {}".format(db_name), "test finishes", "Test completed. Stopping Replicators")
+
+        repl_obj.stop(repl)
+        time.sleep(5)
+        _check_doc_count(db_obj, cbl_db)
+
+        results = query.query_get_docs_limit_offset(cbl_db, limit=query_limit, offset=query_offset)
+
+        return {db_name: doc_ids}
+    except (Exception, RuntimeError) as ex:
+        _log_system_test(thread_name, "exception caught", ex)
+        _log_system_test(thread_name, "quit smoothly")
+    finally:
```

Comment:
We do not need to have as part of this PR. Can we do logs scanning at the end of the system test to check all CBLs and SGW. For SGW we can integrate with eagle eye, but for atleast do logs scanning for all cbls. We can have it as enhancement as next of this

Reply:
have logged a ticket, will be discussing it with server qe team.